torch/testing/_internal/schema_check_mode.py

```diff
@@ -0,0 +1,45 @@
+import torch
+from torch.utils._pytree import tree_flatten
+from torch.fx.operator_schemas import normalize_function
+from torch.testing._internal.jit_utils import clone_inputs
+from torch.utils._python_dispatch import TorchDispatchMode
+
+# This TorchDispatchMode Subclass is used to verify op schemas
+# This TorchDispatchMode Scubclass currently:
+#  - Records the called ops
+#  - Checks for mutations on all inputs
+
+class SchemaCheckMode(TorchDispatchMode):
+    def __init__(self):
+        self.ops = []
+
+    def reset_cache(self):
+        self.ops.clear()
+
+    def display_ops(self):
+        print(*self.ops, sep=",")
+
+    def __torch_dispatch__(self, func, types, args=(), kwargs=None):
+        def has_mutated(before, after):
+            return not torch.equal(before, after) if isinstance(before, torch.Tensor) and isinstance(after, torch.Tensor) else False
```

Comment:
Can we deepcopy before ? (unrelated to refactor)

Reply:
Yes we deepcopy before. (Although I'm not entirely sure if that's what you are asking here)